# HG changeset patch
# User Kai Habel <kai.habel@gmx.de>
# Date 1315495923 -7200
#      Thu Sep 08 17:32:03 2011 +0200
# Node ID 9fc95b9f8001fd391abb47c3b07d317e9bdc97c8
# Parent  cd808de114c18882ca26befbd4cb664db1b5168d
Avoid touching default values.

plot/whitebg.m: Rewrite test to avoid setting of defaultvalues.

diff --git a/scripts/plot/whitebg.m b/scripts/plot/whitebg.m
--- a/scripts/plot/whitebg.m
+++ b/scripts/plot/whitebg.m
@@ -142,26 +142,23 @@ function whitebg (varargin)
       endif
     endif
   endif
 endfunction
 
 %!test
 %! dac = get (0, "defaultaxescolor");
 %! dfc = get (0, "defaultfigurecolor");
-%! set (0, "defaultaxescolor", [1 1 1]);
-%! set (0, "defaultfigurecolor", [1 1 1]);
 %! hf = figure (1232, "visible", "off");
 %! unwind_protect  
 %!   l = line;
-%!   assert (get (hf, "color"), [1 1 1]);
-%!   assert (get (gca, "color"), [1 1 1]);
+%!   assert (get (hf, "color"), dfc);
+%!   assert (get (gca, "color"), dac);
 %!   whitebg;
-%!   assert (get (hf, "color"), [0 0 0]);
-%!   assert (get (gca, "color"), [0 0 0]);
-%!   whitebg([0.2 0.2 0.2])
-%!   assert (get (hf, "color"), [0 0 0]);
-%!   assert (get (gca, "color"), [0.2 0.2 0.2]);
+%!   assert (get (hf, "color"), 1 - dfc);
+%!   assert (get (gca, "color"), 1 - dac);
+%!   c = [0.2 0.2 0.2];
+%!   whitebg (c);
+%!   assert (get (hf, "color"), 1 - dfc);
+%!   assert (get (gca, "color"), c);
 %! unwind_protect_cleanup
 %!   close (hf);
-%!   set (0, "defaultaxescolor", dac);
-%!   set (0, "defaultfigurecolor", dfc);
 %! end_unwind_protect
